datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StackRequestManifester.java

```diff
@@ -402,13 +402,18 @@ void setupInstanceVolumeEncryptionForAzure(StackV4Request stackRequest, Detailed
                 .map(AzureEnvironmentParameters::getResourceEncryptionParameters)
                 .map(AzureResourceEncryptionParameters::getDiskEncryptionSetId);
         if (diskEncryptionSetId.isPresent()) {
+            Optional<String> encryptionKeyUrl = Optional.of(environmentResponse)
+                    .map(DetailedEnvironmentResponse::getAzure)
+                    .map(AzureEnvironmentParameters::getResourceEncryptionParameters)
+                    .map(AzureResourceEncryptionParameters::getEncryptionKeyUrl);
             stackRequest.getInstanceGroups().forEach(ig -> {
                 AzureInstanceTemplateV4Parameters azure = ig.getTemplate().createAzure();
                 AzureEncryptionV4Parameters encryption = azure.getEncryption();
                 if (encryption == null) {
                     encryption = new AzureEncryptionV4Parameters();
                     azure.setEncryption(encryption);
                 }
+                azure.getEncryption().setKey(encryptionKeyUrl.get());
```

Comment:
what if `encryptionKeyUrl` is not present? then we will get an exception here, I think that should be checked, for example in the if condition above, where you are checking `encryption == null` condition

Reply:
`encryptionKeyUrl` is pre-requisite for creation of `diskEncryptionSetId`. Checking `diskEncryptionSetId.isPresent()` is sufficient to ensure that encryptionKeyUrl is non-null/present.

Reply:
same here